### ISSUE TITLE
Fadeout issue

### ISSUE THREAD

**Issue by Heck-R** [pubpub-zz/ppInk#245]
Line styles other than "Stroke" (E.g: "Solid") seem to ignore Alpha settings as far as visibility goes
Not sure if that's intended or not

In case it is not intended, then the issue is precisely that.

In case it is intended, then there are 2 issues:
- It might be just me, but I did not see this anywhere noted, so I did not understand why it did not work
  The closest explanation of this seems to be the following, however since Stroke is being used as a collective word and a style, it's a bit confusing (there is no clear indication of the style having an effect on the separately described fading)
  ```
  - Fading : the stroke drawn will disappear after the defined time(adjustable in options dialogbox/pen tab)
  - Line Style (Stroke/Solid/Dash/Dot/DashDot/DashDotDot) : This will apply the define line style on all type of drawings. Stroke keeps the original drawing which uses the pen pressure to adjust the width. Solid,Dash,... ignore pen pressure.)
  ```
- The styles which cannot visibly fade still have additional disappearance time based on their theoretical alpha, because the disappearance logic continuously changes their theoretical alpha until they are theoretically invisible, and they are only removed at after that
  In practice, the non-stroke styled lines keep being 100% visible for some additional time (about as many milliseconds as the alpha), and then disappeared in an instant
  There is a workaround, which is setting the alpha to 0 when using a non-stroke style, but it's far from obvious at first (I though it's just slow)
  Disabling the Alpha input on the UI, or at least ignoring it at fadeout when an incompatible style is used would be nice

## An additional request:
0 is an invalid fading time value. One can use 0.01 for essentially the same effect, but being able to set 0 would be nice

**Comment by pubpub-zz** [comment]
Hi @Heck-R 
Thanks for reporting the bug. Don't know why this was not set originally and reported yet, but I did the test and it seems to be working that way : can you confirm
About setting fading to 0.01. 0.01 seems to be accepted by the system set using the option. However, first the sampling rate is 15ms and second less than 100ms is very difficult to see.
My understanding of your request is that there was no way to set how fast the fading is performed.I've added a parameter in config.ini for this : `Fading_Decrease` if you increase this parameter you should get what you're looking for

You can try this beta release with both mods.
[ppInk Alpha Dashed.zip](https://github.com/pubpub-zz/ppInk/files/10474056/ppInk.Alpha.Dashed.zip)

Waiting for your feedback before committing

**Comment by Heck-R** [comment]
### Bug
Now it works well for me too, thanks! 🙂

### Request
That's not what I meant
What I meant is I'd like to enter 0 here for no delay. (The 0.01 was just a workaround I used, because it's so short that it felt instant, even if it was 15ms in reality)
![image](https://user-images.githubusercontent.com/29919193/213932708-01c6f087-8d82-4575-b8e8-04e2d62aa58b.png)

### Regarding `Fading_Decrease`
This also crossed my mind, I just didn't mention it yet, (because I did not want to branch off to too many directions) so definitely keep this feature, its pretty nice 😀
Although as a user it would be more convenient to provide the seconds or milliseconds it's still on the screen.
The current setting is okay too, but it's inherently a bit mysterious, so getting the desired fading time is kind of trial and error
I assume its the amount of alpha the stroke looses every tick at the moment, which makes the fading time connected to the initial alpha, which causes the fading time to be dependent on the initial state which is kind of an unexpected behaviour (also, values from 128-254 are then effectively the same most of the time except for 1 moment)

**Comment by pubpub-zz** [comment]
> Now it works well for me too, thanks! 🙂

👍

> That's not what I meant What I meant is I'd like to enter 0 here for no delay. (The 0.01 was just a workaround I used, because it's so short that it felt instant, even if it was 15ms in reality)

I had to extend the mod to prevent some side effect bug but I'm OK to extend range.

> ### Regarding `Fading_Decrease`
> This also crossed my mind, I just didn't mention it yet, (because I did not want to branch off to too many directions) so definitely keep this feature, its pretty nice 😀 Although as a user it would be more convenient to provide the seconds or milliseconds it's still on the screen. The current setting is okay too, but it's inherently a bit mysterious, so getting the desired fading time is kind of trial and error I assume its the amount of alpha the stroke looses every tick at the moment, which makes the fading time connected to the initial alpha, which causes the fading time to be dependent on the initial state which is kind of an unexpected behaviour (also, values from 128-254 are then effectively the same most of the time except for 1 moment)

In order to prevent question/remarks I will keep this parameter accessible only through the  ini file for advance users. 🤫😉

**Comment by pubpub-zz** [comment]
will be closed when next version will be released. Meanwhile the latest test version
[ppInk fading0.zip](https://github.com/pubpub-zz/ppInk/files/10484211/ppInk.fading0.zip)

**Comment by Heck-R** [comment]
Works great, thanks! 🙂

**Comment by pubpub-zz** [comment]
released in 1.9
